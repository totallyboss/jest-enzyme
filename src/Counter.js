import React, { Component } from 'react';

class Counter extends Component {

  state = {
    count: 0,
  };

  makeIncrementer = amount => () => {
    this.setState(prevState => ({
      count: prevState.count + amount,
    }));
  };

  increment = this.makeIncrementer(1);

  decrement = this.makeIncrementer(-1);


  render() {
    const { count } = this.state;

    return(
        <div>
          <p>{count}</p>
          <button className="increment" name="increment" type="button" onClick={this.increment}>+++</button>
          <button className="decrement" name="decrement" type="button" onClick={this.decrement}>---</button>
        </div>
    );
  };

}

export default Counter;
