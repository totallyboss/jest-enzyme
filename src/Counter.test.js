import React from 'react';
import Counter from './Counter';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';


describe('<Counter />', () => {

  it('matches the snapshot', () => {
    const tree = renderer.create(<Counter />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should have the first be zero', () => {
    const wrapper = shallow(<Counter />);
    const count = wrapper.state().count;

    expect(count).toEqual(0);
  });

  it('should incrememnt the count when the button is clicked', () => {
    const wrapper = shallow(<Counter />);
    const incrementBtn = wrapper.find('button.increment');

    incrementBtn.simulate('click');
    const text = wrapper.find('p').text();
    expect(text).toEqual('1');
  });

  it('can decrement the count when the decrement button is clicked', () => {
    const wrapper = shallow(<Counter />);
    const decrementBtn = wrapper.find('button.decrement');

    decrementBtn.simulate('click');
    const text = wrapper.find('p').text();
    expect(text).toEqual('-1');
  });

});
