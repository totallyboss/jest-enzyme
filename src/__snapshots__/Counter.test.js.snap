// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> matches the snapshot 1`] = `
<div>
  <p>
    0
  </p>
  <button
    className="increment"
    name="increment"
    onClick={[Function]}
    type="button"
  >
    +++
  </button>
  <button
    className="decrement"
    name="decrement"
    onClick={[Function]}
    type="button"
  >
    ---
  </button>
</div>
`;
